File: Dante1/Module1.07/Zadanie7.04/main.c
#include <stdio.h>

int sign(int value);

int main()
{
    int input;
    printf("Enter number: ");

    int check = scanf("%i", &input);
    while(check != 1)
    {
        printf("Incorrect input\n");
        return 1;

    }

    int result = sign(input);

    printf("%s\n", (result == 0) ? "zero" : ((result == 1) ? "positive" : "negative"));

    return 0;
}

int sign(int value)
{
    return (value > 0) - (value < 0);
}

File: Dante1/Module1.07/Zadanie7.04/main.rs
fn sign(value: i32) -> i32
{
    return (value > 0) as i32 - (value < 0) as i32;
}

fn main()
{
    let mut input: String = String::new();
    println!("Enter: ");

    std::io::stdin().read_line(&mut input).expect("Incorrect input");
    let number: i32 = input.trim().parse().expect("Incorrect parsing");

    let to_switch: i32 = sign(number);
    if to_switch == 0
    {
        print!("Zero");
    }
    else if to_switch == 1
    {
        print!("Positive");
    }
    else
    {
        print!("Negative");
    }
}
